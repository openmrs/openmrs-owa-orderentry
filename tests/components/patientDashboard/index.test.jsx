import React from 'react';
import expect from 'expect';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import ConnectedPatientDashboard, {
  Dashboard
} from '../../../app/js/components/patientDashboard';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Dashboard {...props} />);
  }
  return mountedComponent;
};

describe('Component: PatientDashboard', () => {
  beforeEach(() => {
    props = {
      location: {
        hash: '#/?patient=6ecsd-jdfn-sdfdf'
      },
      isLoading: false,
      patient: {
        person: {
          personName: {
            givenName: 'Larrystone',
            middleName: 'Katz',
            familyName: 'Larrystone'
          },
          preferredAddress: {
            display: 'My Address'
          },
          gender: 'M',
          age: '19'
        },
        patientIdentifier: {
          identifier: '2001A'
        },
        attributes: [{
          value: '0801000000'
        }]
      },
      note: [{
        value: "Healthy guy",
        auditInfo: {
          creator: {
            display: "admin"
          },
          dateCreated: new Date()
        }
      }],
      showContactInfo: true,
      toggleDetailsView: jest.fn(),
      showContactInfo: true,
      fetchPatientRecord: jest.fn(),
      fetchPatientNote: jest.fn()
    };
    mountedComponent = undefined;
  });

  it('should render properly', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  describe('toggleDetailsView() method', () => {
    it('should call toggleDetailsView()', () => {
      sinon.spy(getComponent().instance(), 'toggleDetailsView');
      getComponent().instance().toggleDetailsView();
      expect(getComponent().instance().toggleDetailsView.calledOnce)
        .toEqual(true);
    });
  });

  describe('Connected PatientDashboard component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        patientReducer: {
          patient: { display: "Amani Hospital" },
        },
        noteReducer: {
          note: []
        },
      });
      const wrapper = shallow(<ConnectedPatientDashboard store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
