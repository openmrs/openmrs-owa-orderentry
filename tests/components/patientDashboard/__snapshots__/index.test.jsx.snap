// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PatientDashboard should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    fetchPatientNote={
      [MockFunction] {
        "calls": Array [
          Array [
            "6ecsd-jdfn-sdfdf",
          ],
        ],
      }
    }
    fetchPatientRecord={
      [MockFunction] {
        "calls": Array [
          Array [
            "6ecsd-jdfn-sdfdf",
          ],
        ],
      }
    }
    isLoading={false}
    location={
      Object {
        "hash": "#/?patient=6ecsd-jdfn-sdfdf",
      }
    }
    note={
      Array [
        Object {
          "auditInfo": Object {
            "creator": Object {
              "display": "admin",
            },
            "dateCreated": 2018-03-22T23:26:15.873Z,
          },
          "value": "Healthy guy",
        },
      ]
    }
    patient={
      Object {
        "attributes": Array [
          Object {
            "value": "0801000000",
          },
        ],
        "patientIdentifier": Object {
          "identifier": "2001A",
        },
        "person": Object {
          "age": "19",
          "gender": "M",
          "personName": Object {
            "familyName": "Larrystone",
            "givenName": "Larrystone",
            "middleName": "Katz",
          },
          "preferredAddress": Object {
            "display": "My Address",
          },
        },
      }
    }
    showContactInfo={true}
    toggleDetailsView={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BreadCrumb
          name="Larrystone. Larrystone"
          patientId={null}
        />,
        <View
          note={
            Array [
              Object {
                "auditInfo": Object {
                  "creator": Object {
                    "display": "admin",
                  },
                  "dateCreated": 2018-03-22T23:26:15.873Z,
                },
                "value": "Healthy guy",
              },
            ]
          }
          patient={
            Object {
              "attributes": Array [
                Object {
                  "value": "0801000000",
                },
              ],
              "patientIdentifier": Object {
                "identifier": "2001A",
              },
              "person": Object {
                "age": "19",
                "gender": "M",
                "personName": Object {
                  "familyName": "Larrystone",
                  "givenName": "Larrystone",
                  "middleName": "Katz",
                },
                "preferredAddress": Object {
                  "display": "My Address",
                },
              },
            }
          }
          showContactInfo={false}
          toggleDetailsView={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "Larrystone. Larrystone",
          "patientId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "note": Array [
            Object {
              "auditInfo": Object {
                "creator": Object {
                  "display": "admin",
                },
                "dateCreated": 2018-03-22T23:26:15.873Z,
              },
              "value": "Healthy guy",
            },
          ],
          "patient": Object {
            "attributes": Array [
              Object {
                "value": "0801000000",
              },
            ],
            "patientIdentifier": Object {
              "identifier": "2001A",
            },
            "person": Object {
              "age": "19",
              "gender": "M",
              "personName": Object {
                "familyName": "Larrystone",
                "givenName": "Larrystone",
                "middleName": "Katz",
              },
              "preferredAddress": Object {
                "display": "My Address",
              },
            },
          },
          "showContactInfo": false,
          "toggleDetailsView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BreadCrumb
            name="Larrystone. Larrystone"
            patientId={null}
          />,
          <View
            note={
              Array [
                Object {
                  "auditInfo": Object {
                    "creator": Object {
                      "display": "admin",
                    },
                    "dateCreated": 2018-03-22T23:26:15.873Z,
                  },
                  "value": "Healthy guy",
                },
              ]
            }
            patient={
              Object {
                "attributes": Array [
                  Object {
                    "value": "0801000000",
                  },
                ],
                "patientIdentifier": Object {
                  "identifier": "2001A",
                },
                "person": Object {
                  "age": "19",
                  "gender": "M",
                  "personName": Object {
                    "familyName": "Larrystone",
                    "givenName": "Larrystone",
                    "middleName": "Katz",
                  },
                  "preferredAddress": Object {
                    "display": "My Address",
                  },
                },
              }
            }
            showContactInfo={false}
            toggleDetailsView={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "Larrystone. Larrystone",
            "patientId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "note": Array [
              Object {
                "auditInfo": Object {
                  "creator": Object {
                    "display": "admin",
                  },
                  "dateCreated": 2018-03-22T23:26:15.873Z,
                },
                "value": "Healthy guy",
              },
            ],
            "patient": Object {
              "attributes": Array [
                Object {
                  "value": "0801000000",
                },
              ],
              "patientIdentifier": Object {
                "identifier": "2001A",
              },
              "person": Object {
                "age": "19",
                "gender": "M",
                "personName": Object {
                  "familyName": "Larrystone",
                  "givenName": "Larrystone",
                  "middleName": "Katz",
                },
                "preferredAddress": Object {
                  "display": "My Address",
                },
              },
            },
            "showContactInfo": false,
            "toggleDetailsView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
